svgast/units.py

```diff
@@ -22,6 +22,66 @@ def __str__(self):
             str_number(self.px / self._px_per_unit),
             self.unit)
 
+    def __abs__(self):
+        return self.px
+
+    def __add__(self, other):
+        if isinstance(other, Length):
+            px = self.px + other.px
+        else:
+            px = self.px + other
+        l = user(px)
+        if type(self) is type(other):
+            return type(self)(l)
+        else:
+            return l
+
+    def __radd__(self, other):
+        assert not isinstance(other, Length)
+        return user(other + self.px)
+
+    def __sub__(self, other):
+        if isinstance(other, Length):
+            px = self.px - other.px
+        else:
+            px = self.px - other
+        l = user(px)
+        if type(self) is type(other):
+            return type(self)(l)
+        else:
+            return l
+
+    def __rsub__(self, other):
+        assert not isinstance(other, Length)
+        return user(other - self.px)
+
+    def __mul__(self, other):
```

Comment:
Multiplication and division are interesting for this:
I think `3 * 2cm` makes sense, but I'm not sure that `3cm * 2mm` does make sense.
I presume you're using complex numbers in the usual way to represent a 2d vector?

Reply:
I've explicitly said that `3cm * 2mm` isn't allowed - only multiplication by a scalar. I read somewhere that `Complex` was the way to say "number" in Python - might be wrong on that front! Happy for suggestions.

Reply:
Erm, there's a numbers.Number whose docstring claims "All numbers inherit from this class.". Complex has a conjugate and stuff and does in fact appear to be a complex number.